### PR TITLE
client: card status: Update visisble card status whenever a clue is given

### LINKED ISSUE #1681: Cluing trash doesn't make it opaque
https://hanab.live/replay/279627/39 the fill should reload to opacity 1 as soon as the clue happens

### PR DESCRIPTION
**Pull request by timotree3**
Should fix #1681. I haven't tested this because playing a game locally isn't working well and due to #1696 I can't open the test case replay

### CODE REVIEW

**Comment by timotree3** [comment]
An alternative solution/follow-up is being discussed in #1681 that I would happily implement.

**Comment by timotree3** [comment]
Okay switched to that other solution. I did locally test this and the client seems to work fine although I can't reproduce the bug on the old revision either.

File: client/src/game/ui/reactive/view/cardsView.ts [hunk 1]
```diff
@@ -110,6 +110,23 @@ const subscribeToCardChanges = (order: number) => {
     location: c.location,
   }), () => updateBorder(order));
 
+  // Card fade and critical indicator
+  subFullState((s) => {
+    const card = s.visibleState!.deck[order];
+    const status = card.suitIndex !== null && card.rank !== null
```
Comment (robertos) [inline client/src/game/ui/reactive/view/cardsView.ts:116]: parens around the condition because the operator precedence of ? is not clear

Comment (timotree3) [reply client/src/game/ui/reactive/view/cardsView.ts:116]: if it's the other grouping than we're &&-ing with a non boolean. I don't think that's possible

Comment (robertos) [reply client/src/game/ui/reactive/view/cardsView.ts:116]: You can definitely && a non-boolean - welcome to javascript :/ Everything casts to everything silently

File: client/src/game/ui/reactive/view/cardsView.ts [hunk 2]
```diff
@@ -110,6 +110,23 @@ const subscribeToCardChanges = (order: number) => {
     location: c.location,
   }), () => updateBorder(order));
 
+  // Card fade and critical indicator
+  subFullState((s) => {
+    const card = s.visibleState!.deck[order];
+    const status = (card.suitIndex !== null && card.rank !== null)
+      ? s.visibleState!.cardStatus[card.suitIndex!][card.rank!]
```
Comment (Zamiell) [inline client/src/game/ui/reactive/view/cardsView.ts:117]: why put the bangs on suitindex and rank?

Comment (timotree3) [reply client/src/game/ui/reactive/view/cardsView.ts:117]: I didn't know how to address the nullability besides this approach. I'm new to TS so any suggestion welcome

File: client/src/game/ui/reactive/view/cardsView.ts (hunk 1 above)
Comment (timotree3) [reply client/src/game/ui/reactive/view/cardsView.ts]: fair. fixed

File: client/src/game/ui/reactive/view/cardsView.ts (hunk 2 above)
Comment (timotree3) [reply client/src/game/ui/reactive/view/cardsView.ts]: Wow I guess I can just remove them. What kind of sorcery is typescript doing??

File: client/src/game/ui/reactive/view/cardsView.ts [hunk 3]
```diff
@@ -110,6 +110,19 @@ const subscribeToCardChanges = (order: number) => {
     location: c.location,
   }), () => updateBorder(order));
 
+  // Card fade and critical indicator
+  subFullState((s) => {
```
Comment (Zamiell) [inline client/src/game/ui/reactive/view/cardsView.ts:114]: if it only uses visible state, then shouldnt the sub be to the visible state?

Comment (robertos) [reply client/src/game/ui/reactive/view/cardsView.ts:114]: It's literally the same

Comment (robertos) [reply client/src/game/ui/reactive/view/cardsView.ts:114]: the "sub visible" is a shortcut for what he did

Comment (Zamiell) [reply client/src/game/ui/reactive/view/cardsView.ts:114]: if its the same, then use the provided helper and put the code in the correct section

Comment (robertos) [reply client/src/game/ui/reactive/view/cardsView.ts:114]: The helper is not in this file.

Comment (Zamiell) [reply client/src/game/ui/reactive/view/cardsView.ts:114]: ok never mind then

**Comment by Zamiell** [comment]
thanks tim for looking into this

**Pull request merged**